modules/bitgo/test/v2/unit/tss/ecdsa.ts

```diff
@@ -0,0 +1,242 @@
+import {
+  readSignedMessage,
+  Ecdsa,
+  ECDSA,
+  ECDSAMethods,
+} from '@bitgo/sdk-core';
+import * as openpgp from 'openpgp';
+import * as should from 'should';
+
+type KeyShare = ECDSA.KeyShare;
+const encryptNShare = ECDSAMethods.encryptNShare;
+const createCombinedKey = ECDSAMethods.createCombinedKey;
+
+describe('test tss helper functions', function () {
+  let mpc: Ecdsa;
+
+  let userKeyShare: KeyShare;
+  let backupKeyShare: KeyShare;
+  let bitgoKeyShare: KeyShare;
+
+  let userKey;
+  let backupKey;
+  let bitgoKey;
+
+  let userGpgKeypair: { publicKey: string, privateKey: string };
+  let backupGpgKeypair: { publicKey: string, privateKey: string };
+  let bitgoGpgKeypair: { publicKey: string, privateKey: string };
+
+  let commonKeychain: string;
+
+  before(async function () {
+    mpc = new Ecdsa();
+    this.timeout(50000);
+
+    userKeyShare = await mpc.keyShare(1, 2, 3);
+    backupKeyShare = await mpc.keyShare(2, 2, 3);
+    bitgoKeyShare = await mpc.keyShare(3, 2, 3);
+
+    userKey = mpc.keyCombine(userKeyShare.pShare, [backupKeyShare.nShares[1], bitgoKeyShare.nShares[1]]);
+    backupKey = mpc.keyCombine(backupKeyShare.pShare, [userKeyShare.nShares[2], bitgoKeyShare.nShares[2]]);
+    bitgoKey = mpc.keyCombine(bitgoKeyShare.pShare, [backupKeyShare.nShares[3], userKeyShare.nShares[3]]);
+    
+    (userKey.xShare.y + userKey.xShare.chaincode).should.equal(backupKey.xShare.y + backupKey.xShare.chaincode);
+    (userKey.xShare.y + userKey.xShare.chaincode).should.equal(bitgoKey.xShare.y + bitgoKey.xShare.chaincode);
+    commonKeychain = userKey.xShare.y + userKey.xShare.chaincode;
+
+    userGpgKeypair = await openpgp.generateKey({
+      userIDs: [
+        {
+          name: 'user',
+          email: 'user@bitgo.com',
+        },
+      ],
+    });
+    backupGpgKeypair = await openpgp.generateKey({
+      userIDs: [
+        {
+          name: 'backup',
+          email: 'backup@bitgo.com',
+        },
+      ],
+    });
+    bitgoGpgKeypair = await openpgp.generateKey({
+      userIDs: [
+        {
+          name: 'bitgo',
+          email: 'bitgo@bitgo.com',
+        },
+      ],
+    });
+  });
+
+  describe('encryptNShare', function () {
+    it('should encrypt n share', async function () {
+      for (let i = 2; i <= 3; i++) {
+        const encryptedNShare = await ECDSAMethods.encryptNShare({
```

Comment:
Not exactly sure why this loop for checking for i=2, 3 when the `recipientGpgPublicArmor` is always the same?, are we testing cases where we are sharing the NShare from the user to backup and bitgo ?

Reply:
Was this resolved?